src/test/kotlin/org/rust/lang/refactoring/RsImportOptimizerTest.kt

```diff
@@ -232,7 +289,78 @@ class RsImportOptimizerTest: RsTestBase() {
         }
     """)
 
-    private fun doTest(@Language("Rust") code: String, @Language("Rust") excepted: String){
+    @ProjectDescriptor(WithStdlibAndDependencyRustProjectDescriptor::class)
+    fun `test group imports by semantics`() = doTest("""
+        extern crate dep_lib_target;
+
+        use bbb::{fff, eee};
+        use aaa::bbb::ccc;
+        use std::string;
+        use std::{mem, io};
+        use dep_lib_target;
+        fn main() {}
+    """, """
+        extern crate dep_lib_target;
+
+        use std::{io, mem};
+        use std::string;
+
+        use dep_lib_target;
+
+        use aaa::bbb::ccc;
+        use bbb::{eee, fff};
+
+        fn main() {}
+    """)
+
+    @ProjectDescriptor(WithStdlibRustProjectDescriptor::class)
+    fun `test group imports by semantics with mod`() = doTest("""
+        use bbb::{fff, eee};
+        use super::ccc::bbb;
+        use aaa::bbb;
+        use crate::bbb::ccc;
+        use crate::aaa::bbb;
+        use self::aaa;
+        use sss::aaa;
+        use std::{mem, string, io};
+        mod aaa {
+            use bbb::*;
+            use xxx::yyy;
+            use super::aaa::bbb::ccc;
+            use std::io;
+            pub mod bbb {
+                pub mod ccc {}
+            }
+        }
+    """, """
+        use std::{io, mem, string};
+
+        use aaa::bbb;
+        use bbb::{eee, fff};
+        use sss::aaa;
+
+        use crate::aaa::bbb;
+        use crate::bbb::ccc;
+
+        use super::ccc::bbb;
+
+        use self::aaa;
```

Comment:
Not sure that we should add empty line between `self`, `crate`, `super` and other local imports
@matklad what do you think?

Reply:
I don't have a strong opinion and mildly in favor: `self` and `super` imports are pretty rare in practice, and when they are used, it is ok to slightly emphasize them.